### PR TITLE
Add support for contentOffset prop in scrollview

### PR DESCRIPTION
**Pull request by pprabhu-ng**
- update values for prop support in native prop config
- add support for contentOffset prop
- cleanup redundant checks

### CODE REVIEW

File: ReactSkia/LegacyNativeModules/uimanager/UiManagerModule.cpp [hunk 1]
```diff
@@ -121,25 +121,27 @@ dynamic Uimanager::getConstantsForViewManager(std::string viewManagerName) {
...
+      "nestedScrollEnabled", false)(
+      "onMomentumScrollBegin", false)("onMomentumScrollEnd", false)("onScroll", true)("onScrollBeginDrag", false)("onScrollEndDrag", false)("onScrollToTop", false)("overScrollMode", false)(
+      "pagingEnabled", true)("persistentScrollbar", true)("pinchGestureEnabled",false)(
+      "scrollEnabled", true)("scrollEventThrottle", true)("scrollToOverflowEnabled", false)("scrollsToTop", false)("sendMomentumEvents", false)("showsHorizontalScrollIndicator", true)(
       "showsVerticalScrollIndicator", true)("snapToAlignment", true)("snapToEnd", true)("snapToInterval", true)("snapToOffsets", true)("snapToStart", true)(
       "zoomScale", true);
     auto bubblingEventTypes = folly::dynamic::object();
-    auto directEventTypes = folly::dynamic::object("topScrollToTop", folly::dynamic::object("registrationName", "onScrollToTop"));
+    auto directEventTypes = folly::dynamic::object("topContentSizeChange",folly::dynamic::object("registrationName","onContentSizeChange"))(
+      "topScroll",folly::dynamic::object("registrationName","onScroll"));
```
Comment (chatrapathi06) [inline ReactSkia/LegacyNativeModules/uimanager/UiManagerModule.cpp]: onscroll,onContentSizeChange  should be the view direct event type.

File: ReactSkia/components/RSkComponentScrollView.cpp [hunk 2]
```diff
@@ -477,12 +439,28 @@ ScrollStatus RSkComponentScrollView::handleScroll(rnsKey direction,SkIRect candi
...
+  return handleScroll(scrollPos,isFlushDisplay);
+}
+
+ScrollStatus RSkComponentScrollView::handleScroll(SkPoint scrollPos, bool isFlushDisplay) {
 
   RnsShell::ScrollLayer* scrollLayer= SCROLL_LAYER_HANDLE;
   if(scrollPos == scrollLayer->getScrollPosition()) return noScroll;
 
-  scrollLayer->client().notifyFlushBegin();
+  if(isFlushDisplay) scrollLayer->client().notifyFlushBegin();
```
Comment (chatrapathi06) [inline ReactSkia/components/RSkComponentScrollView.cpp:463]: indentation fix

File: ReactSkia/components/RSkComponentScrollView.cpp [hunk 3]
```diff
@@ -491,7 +469,7 @@ ScrollStatus RSkComponentScrollView::handleScroll(SkPoint scrollPos) {
 #endif
 
   scrollLayer->invalidate(LayerPaintInvalidate);
-  scrollLayer->client().notifyFlushRequired();
+  if(isFlushDisplay) scrollLayer->client().notifyFlushRequired();
```
Comment (chatrapathi06) [inline ReactSkia/components/RSkComponentScrollView.cpp:472]: indentation fix

File: ReactSkia/LegacyNativeModules/uimanager/UiManagerModule.cpp (hunk 1 above)
Comment (pprabhu-ng) [reply ReactSkia/LegacyNativeModules/uimanager/UiManagerModule.cpp]: updated.

**Review by chatrapathi06 — APPROVED** [review]
Approved

**Review by munezbn — APPROVED** [review]

**Pull request merged**